domain/prefixmanager/prefix/prefix.go

```diff
@@ -0,0 +1,46 @@
+package prefix
+
+import "github.com/pkg/errors"
+
+const (
+	prefixZero byte = 0
+	prefixOne  byte = 1
+)
+
+// Prefix is a database prefix that is used to manage more than one database at once.
+type Prefix struct {
+	value byte
+}
+
+// Serialize serializes the prefix into a byte slice
+func (p *Prefix) Serialize() []byte {
+	return []byte{p.value}
+}
+
+// Equal returns whether p equals to other
+func (p *Prefix) Equal(other *Prefix) bool {
+	return p.value == other.value
+}
+
+// Flip returns the opposite of the current prefix
+func (p *Prefix) Flip() *Prefix {
+	value := prefixZero
```

Comment:
Optional: All this can be written in one line 
```
return &Prefix {value: 1-p.value }
```

Reply:
I think it's clearer the way it is